### PR TITLE
refactor: replace ColumnOption::DialectSpecific with ColumnOption::Comment

### LINKED ISSUE #158: Refactor parsing sql comment
**Description**

Replace `ColumnOption::DialectSpecific` with `ColumnOption::Comment` to parse sql comment.

**Proposal**

ref: https://github.com/CeresDB/ceresdb/blob/8d1472a33054ac7f92cc957a30f1a83e6da005f6/sql/src/parser.rs#L476

**Additional context**
ref #154

### PR DESCRIPTION
**Pull request by baojinri**
# Which issue does this PR close?

Closes #158 

# Rationale for this change
 simplify code when parse "comment"

# What changes are included in this PR?
replace ColumnOption::DialectSpecific with ColumnOption::Comment when parse "comment"

# Are there any user-facing changes?
None

# How does this change test

- tested by the compiler
- existing UT

### CODE REVIEW

**Comment by ShiKaiWi** [comment]
@baojinri Would you add a test in the `tests` for creating table with comment?

**Review by chunshao90 — APPROVED** [review]
LGTM

**Review by ShiKaiWi — CHANGES_REQUESTED** [review]
Add a test case for comment in the `tests`.

File: sql/src/parser.rs [hunk 1]
```diff
@@ -624,6 +637,22 @@ mod tests {
         });
         expect_parse_ok(sql, expected).unwrap();
 
+        // positive case, multiple columns with comment
+        let sql = "CREATE TABLE mytbl(c1 timestamp, c2 double comment 'id', c3 string comment 'name',) ENGINE = XX";
```
Comment (waynexia) [inline sql/src/parser.rs:641]: This case is good, can you also add one in our integration test suit https://github.com/CeresDB/ceresdb/blob/main/tests/cases/local/05_ddl/create_tables.sql?

**Review by waynexia — APPROVED** [review]

**Review by ShiKaiWi — APPROVED** [review]
Well done!

**Pull request merged**